lib/active_merchant/billing/gateways/shift4.rb

```diff
@@ -0,0 +1,290 @@
+        @client_guid = options[:client_guid]
+        @auth_token = options[:auth_token]
+        super
+        @access_token = setup_access_token
+      end
+
+      def purchase(money, authorization, options = {})
+        post = {}
+        add_datetime(post, options)
+        add_invoice(post, money, options)
+        add_clerk(post, options)
+        add_transaction(post, options)
+        add_card(post, authorization, options)
+
+        commit('sale', post, options)
+      end
+
+      def authorize(money, card, options = {})
+        post = {}
+        add_datetime(post, options)
+        add_invoice(post, money, options)
+        add_clerk(post, options)
+        add_transaction(post, options)
+        add_card(post, card, options)
+        add_customer(post, options)
+
+        commit('authorization', post, options)
+      end
+
+      def capture(money, authorization, options = {})
+        post = {}
+        add_datetime(post, options)
+        add_invoice(post, money, options)
+        add_clerk(post, options)
+        add_transaction(post, options)
+        add_card(post, authorization, options)
+
+        commit('capture', post, options)
+      end
+
+      def refund(money, authorization, options = {})
+        post = {}
+        add_datetime(post, options)
+        add_invoice(post, money, options)
+        add_clerk(post, options)
+        add_transaction(post, options)
+        add_customer(post, options)
+        add_card(post, authorization, options)
+
+        commit('refund', post, options)
+      end
+
+      def void(options = {})
+        post = {}
+        commit('invoice', post, options)
+      end
+
+      def verify(credit_card, options = {})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { authorize(100, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r(("Number\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+          gsub(%r(("expirationDate\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+          gsub(%r(("FirstName\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]').
+          gsub(%r(("LastName\\?"\s*:\s*\\?")[^"]*)i, '\1[FILTERED]')
+      end
+
+      private
+
+      def add_credentials(post, options)
+        post[:dateTime] = current_date_time
+        post[:credential] = {}
+        post[:credential][:clientGuid] = @client_guid
+        post[:credential][:authToken] = @auth_token
+      end
+
+      def setup_access_token
+        post = {}
+        add_credentials(post, options)
+
+        response = commit('accesstoken', post, request_headers(options))
+        response.params['result'][0]['credential']['accessToken']
+      end
+
+      def add_clerk(post, options)
+        post[:clerk] = {}
+        post[:clerk][:numericId] = options[:numeric_id]
```

Comment:
I agree with @ajawadmirza earlier comment about changing `options[:numeric_id]` too `options[:clerk_id]`. I think it'll make more sense when we expose this field to end uses.